### PR TITLE
Update Xinmeitulu & Meitua.top: Support filter/search & translate

### PR DESCRIPTION
**Pull request by cuong-tran**
Checklist:

- [X] Updated `extVersionCode` value in `build.gradle` for individual extensions
- [X] Updated `overrideVersionCode` or `baseVersionCode` as needed for all multisrc extensions
- [X] Referenced all related issues in the PR body (e.g. "Closes #xyz")
- [X] Added the `isNsfw = true` flag in `build.gradle` when appropriate
- [X] Have not changed source names
- [X] Have explicitly kept the `id` if a source's name or language were changed
- [X] Have tested the modifications by compiling and running the extension through Android Studio
- [X] Have removed `web_hi_res_512.png` when adding a new extension

### CODE REVIEW

File: src/all/meituatop/src/eu/kanade/tachiyomi/extension/all/meituatop/MeituaTop.kt [hunk 1]
```diff
@@ -95,14 +95,22 @@ class MeituaTop : HttpSource() {
...
 
-    private class RegionFilter : Filter.Select<String>(
+    private class RegionFilter(regionList: Array<String>) : Filter.Select<String>(
         "Region",
-        arrayOf("All", "国产美女", "韩国美女", "台湾美女", "日本美女", "欧美美女", "泰国美女"),
+        regionList,
     )
 
+    private fun getRegionList(): Array<String> {
+        return if (Locale.getDefault().equals("zh")) {
```
Comment (stevenyomi) [inline src/all/meituatop/src/eu/kanade/tachiyomi/extension/all/meituatop/MeituaTop.kt]: You should translate the `SManga.genre` field too.

**Review by stevenyomi — CHANGES_REQUESTED** [review]

**Review by AwkwardPeak7 — DISMISSED** [review]

File: src/all/xinmeitulu/src/eu/kanade/tachiyomi/extension/all/xinmeitulu/Xinmeitulu.kt [hunk 2]
```diff
@@ -88,6 +113,45 @@ class Xinmeitulu : ParsedHttpSource() {
 
     override fun imageUrlParse(document: Document): String = throw UnsupportedOperationException()
 
+    // Filters
+
+    override fun getFilterList(): FilterList {
+        return FilterList(
+            RegionFilter(getRegionList()),
```
Comment (stevenyomi) [inline src/all/xinmeitulu/src/eu/kanade/tachiyomi/extension/all/xinmeitulu/Xinmeitulu.kt:139]: Translate this too.

**Review by stevenyomi — CHANGES_REQUESTED** [review]

File: src/all/xinmeitulu/src/eu/kanade/tachiyomi/extension/all/xinmeitulu/Xinmeitulu.kt (hunk 2 above)
Comment (cuong-tran) [reply src/all/xinmeitulu/src/eu/kanade/tachiyomi/extension/all/xinmeitulu/Xinmeitulu.kt:139]: It's already translated:
```
    private fun getRegionList(): Array<Pair<String?, String>> {
        return if (Locale.getDefault().equals("zh")) {
            arrayOf(
                null to "全部",
                "zhongguodalumeinyu" to "中国大陆美女",
                "taiguomeinyu" to "泰国美女",
                "ribenmeinyu" to "日本美女",
                "hanguomeinyu" to "韩国美女",
                "taiwanmeinyu" to "台湾美女",
                "oumeimeinyu" to "欧美美女",
            )
        } else {
            arrayOf(
                null to "All",
                "zhongguodalumeinyu" to "Chinese beauty",
                "taiguomeinyu" to "Thailand beauty",
                "ribenmeinyu" to "Japanese beauty",
                "hanguomeinyu" to "Korean beauty",
                "taiwanmeinyu" to "Taiwanese beauty",
                "oumeimeinyu" to "European & American beauty",
            )
        }
    }
```

Comment (stevenyomi) [reply src/all/xinmeitulu/src/eu/kanade/tachiyomi/extension/all/xinmeitulu/Xinmeitulu.kt:139]:
```kotlin
    override fun popularMangaFromElement(element: Element) = SManga.create().apply {
        setUrlWithoutDomain(element.select("figure > a").attr("abs:href"))
        title = element.select("figcaption").text()
        thumbnail_url = element.select("img").attr("abs:data-original-")
        genre = element.select("a.tag").joinToString(", ") { it.text() }  // <--- THIS
    }
```

Comment (cuong-tran) [reply src/all/xinmeitulu/src/eu/kanade/tachiyomi/extension/all/xinmeitulu/Xinmeitulu.kt:139]: Function 'popularMangaFromElement' only query list of popular mangas, which won't use 'genre'.

Also the 'tag' which the site's using are completely various, much like hashtag than genre. They are not a fixed list so it's unable to translate them.

Comment (stevenyomi) [reply src/all/xinmeitulu/src/eu/kanade/tachiyomi/extension/all/xinmeitulu/Xinmeitulu.kt:139]: Do `removeSuffix("写真")` and the last tag of each entry can be translated.

**Comment by cuong-tran** [comment]
Added genre translation, but again, genre won't show in manga browse list & it's not available in manga's detail page. Is there anyway to pass `SManga` from `popularMangaFromElement` to `mangaDetailsParse`?

**Comment by stevenyomi** [comment]
> Added genre translation, but again, genre won't show in manga browse list & it's not available in manga's detail page. Is there anyway to pass `SManga` from `popularMangaFromElement` to `mangaDetailsParse`?

Oh right. Probably open an issue in Mihon lol.

**Review by stevenyomi — DISMISSED** [review]

**Comment by cuong-tran** [comment]
Don't merge this yet. I'm doing more translation.

File: src/all/meituatop/src/eu/kanade/tachiyomi/extension/all/meituatop/MeituaTop.kt [hunk 3]
```diff
@@ -95,17 +95,43 @@ class MeituaTop : HttpSource() {
...
+    }
+
     private fun String.pageNumber() = numberRegex.findAll(this).last().value.toInt()
 
     private val numberRegex by lazy { Regex("""\d+""") }
 
     private val dateFormat by lazy { SimpleDateFormat("yyyy-MM-dd", Locale.ENGLISH) }
+
+    private fun translate(chinese: String): String {
+        if (Locale.getDefault().equals("zh")) return chinese
```
Comment (stevenyomi) [inline src/all/meituatop/src/eu/kanade/tachiyomi/extension/all/meituatop/MeituaTop.kt]:
```suggestion
        if (Locale.getDefault().getLanguage() == "zh") return chinese
```

File: src/all/xinmeitulu/src/eu/kanade/tachiyomi/extension/all/xinmeitulu/Xinmeitulu.kt [hunk 4]
```diff
@@ -66,10 +92,28 @@ class Xinmeitulu : ParsedHttpSource() {
 
     override fun mangaDetailsParse(document: Document) = SManga.create().apply {
         setUrlWithoutDomain(document.selectFirst("link[rel=canonical]")!!.attr("abs:href"))
-        title = document.select(".container > h1").text()
-        description = document.select(".container > *:not(div)").text()
+        title = document.select(".container > h1").text().trim()
```
Comment (stevenyomi) [inline src/all/xinmeitulu/src/eu/kanade/tachiyomi/extension/all/xinmeitulu/Xinmeitulu.kt]:
```suggestion
        title = document.select(".container > h1").text()
```
Jsoup already trims the text.

File: src/all/xinmeitulu/src/eu/kanade/tachiyomi/extension/all/xinmeitulu/Xinmeitulu.kt [hunk 5]
```diff
@@ -88,6 +132,60 @@ class Xinmeitulu : ParsedHttpSource() {
...
+            "taiguomeinyu" to translate("泰国美女"),
+            "ribenmeinyu" to translate("日本美女"),
+            "hanguomeinyu" to translate("韩国美女"),
+            "taiwanmeinyu" to translate("台湾美女"),
+            "oumeimeinyu" to translate("欧美美女"),
+        )
+    }
+
+    private fun translate(it: String): String {
+        if (Locale.getDefault().equals("zh")) return it
```
Comment (stevenyomi) [inline src/all/xinmeitulu/src/eu/kanade/tachiyomi/extension/all/xinmeitulu/Xinmeitulu.kt]: Same here.

File: src/all/xinmeitulu/src/eu/kanade/tachiyomi/extension/all/xinmeitulu/Xinmeitulu.kt [hunk 6]
```diff
@@ -88,6 +132,60 @@ class Xinmeitulu : ParsedHttpSource() {
...
+            "全部" -> "All"
+            "中国大陆美女" -> "Chinese beauty"
+            "泰国美女" -> "Thailand beauty"
+            "日本美女" -> "Japanese beauty"
+            "韩国美女" -> "Korean beauty"
+            "台湾美女" -> "Taiwanese beauty"
+            "欧美美女" -> "European & American beauty"
+            // Descriptions
+            "拍摄机构" -> "Studio"
+            "相关编号" -> "Related"
```
Comment (stevenyomi) [inline src/all/xinmeitulu/src/eu/kanade/tachiyomi/extension/all/xinmeitulu/Xinmeitulu.kt]: Should be something like issue number.

**Review by stevenyomi — CHANGES_REQUESTED** [review]

**Comment by beer-psi** [comment]
Closing because you seem to not have a good enough command of the Chinese language and are resorting to machine translation. Feel free to maintain the English versions on your own.

**Pull request closed without merging**